### PR TITLE
Remove extra boot classpath entries for retrolambda

### PR DESCRIPTION
**Pull request by kageiit**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -45,6 +45,10 @@ def addCommonConfigurationForJavaModules(Project project) {
 }
 
 def addCommonConfigurationForAndroidModules(Project project) {
+    project.dependencies {
+        provided files("lib/rt-stub.jar")
```
Comment (seanabraham) [inline build.gradle]: Do you want to always add this or add it only when retrolambda is present?

File: dependencies.gradle [hunk 2]
```diff
@@ -9,7 +9,7 @@ def build = [
         androidPlugin    : 'com.android.tools.build:gradle:2.2.2',
         androidAptPlugin : 'com.neenbedankt.gradle.plugins:android-apt:1.8',
         butterKnifePlugin: "com.jakewharton:butterknife-gradle-plugin:${versions.butterKnifeVersion}",
-        retrolambdaPlugin: 'me.tatarka:gradle-retrolambda:3.3.0',
+        retrolambdaPlugin: 'me.tatarka:gradle-retrolambda:3.2.5',
```
Comment (seanabraham) [inline dependencies.gradle]: Why the downgrade?

Comment (kageiit) [inline dependencies.gradle]: oops, by mistake. will revert

File: build.gradle (hunk 1 above)
Comment (kageiit) [inline build.gradle]: this is just the test project. I will clean this up in the next change

Comment (kageiit) [inline build.gradle]: next change will add this automagically to provided deps in android rules if retrolambda is enabled

File: buildSrc/src/main/groovy/com/uber/okbuck/generator/RetroLambdaGenerator.groovy [hunk 3]
```diff
@@ -22,7 +23,11 @@ final class RetroLambdaGenerator {
         String outputText = output.text
         outputText = outputText.replaceAll('gen-dir', target.rootProject.file(BuckConstants.DEFAULT_BUCK_OUT_GEN_PATH).absolutePath)
                 .replaceAll('retrolambda-jar', target.retroLambdaJar)
-        outputText = outputText.replaceAll('android-jar', target.bootClasspath)
+        if (target instanceof AndroidTarget) {
+            outputText = outputText.replaceAll('android-jar', target.bootClasspath ? target.bootClasspath : "")
```
Comment (seanabraham) [inline buildSrc/src/main/groovy/com/uber/okbuck/generator/RetroLambdaGenerator.groovy]: Nit pick but you can replace `target.bootClasspath ? target.bootClasspath : ""` with `target.bootClasspath ?: ""` I believe

File: build.gradle (hunk 1 above)
Comment (seanabraham) [inline build.gradle]: Ah right I missed that this was just the build.gradle file for one of the sample projects.

File: buildSrc/src/main/groovy/com/uber/okbuck/generator/RetroLambdaGenerator.groovy (hunk 3 above)
Comment (kageiit) [inline buildSrc/src/main/groovy/com/uber/okbuck/generator/RetroLambdaGenerator.groovy]: sure

**Pull request merged**